cocoa/vendor/bugsnag-cocoa/Source/BugsnagSessionTracker.m

```diff
@@ -96,7 +96,7 @@ - (void)send {
         }
         BugsnagSessionTrackingPayload *payload = [[BugsnagSessionTrackingPayload alloc] initWithSessions:sessions];
 
-        if (payload.sessions.count > 0) {
+        if (payload.sessions.count > 0 && self.config.apiKey.length > 0) {
```

Comment:
This check (and the one in `getBodyFromReports`) should instead be much earlier in the process of sending requests. Calls to the public interface (in `Bugsnag.h`) and automatic events from lifecycle callbacks should check if API key is nil and no-op if that is the case. `+[Bugsnag bugsnagStarted]` could potentially be beefed up a bit to check both whether `notifier` and API key are nil.

Reply:
Well yeah, API key should be not be nullable and initializing with a configuration object should require an API key.